game/outfits.rpy

```diff
@@ -0,0 +1,304 @@
+        utils.TIME_BLOCK_MID_MORNING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_LATE_MORNING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_AFTERNOON: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_EVENING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_NIGHT: DEFAULT_OUTFIT_CASUAL_WEEKEND
+    }
+
+    DEFAULT_OUTFIT_SCHEDULE_WEEKDAY_LOW_AFFINITY = {
+        utils.TIME_BLOCK_EARLY_MORNING: DEFAULT_OUTFIT_UNIFORM,
+        utils.TIME_BLOCK_MID_MORNING: DEFAULT_OUTFIT_UNIFORM,
+        utils.TIME_BLOCK_LATE_MORNING: DEFAULT_OUTFIT_UNIFORM,
+        utils.TIME_BLOCK_AFTERNOON: DEFAULT_OUTFIT_UNIFORM,
+        utils.TIME_BLOCK_EVENING: DEFAULT_OUTFIT_UNIFORM,
+        utils.TIME_BLOCK_NIGHT: DEFAULT_OUTFIT_UNIFORM
+    }
+
+    DEFAULT_OUTFIT_SCHEDULE_WEEKEND_LOW_AFFINITY = {
+        utils.TIME_BLOCK_EARLY_MORNING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_MID_MORNING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_LATE_MORNING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_AFTERNOON: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_EVENING: DEFAULT_OUTFIT_CASUAL_WEEKEND,
+        utils.TIME_BLOCK_NIGHT: DEFAULT_OUTFIT_CASUAL_WEEKEND
+    }
+
+    def set_outfit(outfit):
+        """
+        Sets Natsuki's appearance using the given outfit.
+        """
+        global current_outfit_name
+        
+        if not isinstance(outfit, Outfit):
+            raise Exception("Outfit given is not an Outfit-class object")
+            return
+
+        if outfit.clothes is None:
+            raise Exception("Outfit clothing cannot be None")
+            return
+
+        if outfit.hairstyle is None:
+            raise Exception("Outfit hairstyle cannot be None")
+            return
+
+        if outfit.unlocked:
+            store.persistent.jn_natsuki_current_outfit = outfit.clothes
+            store.persistent.jn_natsuki_current_hairstyle = outfit.hairstyle
+            store.persistent.jn_natsuki_current_accessory = outfit.accessory
+            store.persistent.jn_natsuki_current_eyewear = outfit.eyewear
+            store.persistent.jn_natsuki_current_headgear = outfit.headgear
+            store.persistent.jn_natsuki_current_necklace = outfit.necklace
+            current_outfit_name = outfit.reference_name
+
+        else:
+            utils.log("Cannot dress Natsuki in outfit {0}; outfit is locked".format(outfit.name))
+
+    def get_outfit_for_time_block():
+        """
+        Returns the outfit corresponding to affinity, the current time block and whether or not is is a weekday.
+        """
+        if jn_affinity.get_affinity_state() >= jn_affinity.AFFECTIONATE:
+            if utils.get_is_weekday():
+                return DEFAULT_OUTFIT_SCHEDULE_WEEKDAY_HIGH_AFFINITY.get(utils.get_current_time_block())
+
+            else:
+                return DEFAULT_OUTFIT_SCHEDULE_WEEKEND_HIGH_AFFINITY.get(utils.get_current_time_block())
```

Comment:
Looking at this, I think you may be able to make a class for outfit groups. Then we can use `.get_weekday_clothes()` and `.get_weekend_clothes()` on it internally, which would automatically pull from a timeblock

Would really cut down on this logic

Reply:
Could do, but given we plan on re-writing how outfits work is this worthwhile?